### DISCUSSION TITLE
how app()->bindMethod() works?

### DISCUSSION THREAD

**Discussion by aimonext** [laravel/framework#48233] [Q&A] [answered]
How does this actually work? if not then how i can access my custom method like app()->test(), Is it possible?

![image](https://github.com/laravel/framework/assets/110348549/b06672d5-245b-46cb-8d37-edf9a35824a9)

**Comment by henzeb** [comment, marked as answer]
That's not how it supposed to work. Maybe this example might illuminate things for you:

```php
class Hello {
    public function say(string $something = null): void {
        dump('Hello '.($something ?? 'space'));
    }
}

(new Hello())->say();

(new Hello())->say('humans');

app()->bindMethod([Hello::class, 'say'], function (Hello $instance, $app) {
    // Custom implementation for the process() method
    return $instance->say('world');
});

if(app()->hasMethodBinding(Hello::class.'@say')) {
    app()->callMethodBinding(Hello::class.'@say', new Hello());
}
```

A reallife example in laravel would be: https://laravel.com/docs/10.x/queues#handle-method-dependency-injection

It's used to override the dependency injection on methods at certain places. In the queue you can tell it to handle injection differently for certain jobs
